Add Attendee API endpoints (GET, POST, DELETE)

shreyarana7 opened the pull request:
Added app/db/attendees.py for Core operations
Added app/service/attendees.py
Updated app/models/attendees.py
Updated app/routes/attendees.py

JaviEst reviewed (changes requested):
I have not looked yet into the pr, but one think I see that is missing from start is that there are no tests ...

Please look into adding some testing for all endpoints. You can see plenty of examples from the users and events tests

shreyarana7 commented:
Added app/db/attendees.py for Core operations
Added app/service/attendees.py
Updated app/models/attendees.py
Updated app/routes/attendees.py
Added Test Cases

JaviEst commented on code/backend/app/models/attendees.py:
```diff
@@ -1,37 +1,58 @@
+    @classmethod
+    def validate_status(cls, v: AttendeeStatus | None) -> AttendeeStatus:
+        """Default to RSVPed if no status is given."""
+        return v or AttendeeStatus.RSVPED
```
We should default to NULL not RSVPED. Because the entry could be created before anyone has replied to the invitation

JaviEst commented on code/backend/app/models/attendees.py:
```diff
@@ -1,37 +1,58 @@
-Human code: 100%
-
+AI-generated code: 70%
+Human code: 30%
```
Nit: add new line below

JaviEst commented on code/backend/app/routes/attendees.py:
```diff
@@ -1,72 +1,121 @@
 )
 async def delete_attendee(attendee_id: UUID) -> models_attendees.AttendeeRead:
-    raise HTTPException(status_code=501, detail="Not implemented")
+    return await attendees_service.delete_attendee_service(attendee_id)
```
Nit: add new line at the end of the file please

shreyarana7 replied:
I added lines

JaviEst commented on code/backend/app/service/attendees.py:
```diff
@@ -0,0 +1,135 @@
+        raise HTTPException(status_code=500, detail="Internal server error") from e
+    except Exception as e:
+        logger.error(f"Unexpected error while deleting attendee: {str(e)}")
+        raise HTTPException(status_code=500, detail="Internal server error") from e
```
Nit: add new line at the end of the file please

JaviEst commented on code/backend/test/test_attendees_failure_cases.py:
```diff
@@ -0,0 +1,227 @@
+    )
+
+    # Intentional wrong expectation:
+    assert r.status_code == 404, f"Expected 404 to demo failure, got {r.status_code}"
```
Nit: add new line at the end of the file please

JaviEst commented on code/backend/test/test_attendees_success_cases.py:
```diff
@@ -0,0 +1,187 @@
+        params={"filter_expression": f"event_id:eq:{event_id}", "offset": 0, "limit": 50},
+    )
+    assert lst2.status_code == 200
+    assert all(i["attendee_id"] != str(att_id) for i in lst2.json()["items"])
```
Nit: add new line at the end of the file please

JaviEst commented on code/backend/app/service/attendees.py:
```diff
@@ -0,0 +1,135 @@
+"""
+AI-generated code: 80%
+Human code: 20%
```
Nit: add new line below

JaviEst commented on code/backend/test/test_attendees_failure_cases.py:
```diff
@@ -0,0 +1,227 @@
+        await engine.dispose()
+        try:
+            os.remove(path)
+        except PermissionError:
```
No need to add the comment

JaviEst commented on code/backend/test/test_attendees_failure_cases.py:
```diff
@@ -0,0 +1,222 @@
+    assert r.status_code == 404
+    detail = r.json().get("detail", "")
+    assert detail and "no such" in detail.lower()
+
```
These 2 tests seem redundant. I think we only need one?

JaviEst commented on code/backend/test/test_attendees_success_cases.py:
```diff
@@ -0,0 +1,187 @@
+
+    # point all module engines at the same test engine
+    db.engine = engine
+    db.metadata = attendees_md
```
+1

JaviEst commented on code/backend/app/models/attendees.py:
```diff
@@ -1,37 +1,58 @@
-
-Human code: 100%
-
+AI-generated code: 70%
```
+1

JaviEst commented on code/backend/app/models/attendees.py:
```diff
@@ -1,37 +1,58 @@
+    items: list[AttendeeRead]
+    total: int
+    offset: int
+    limit: int
```
+1

JaviEst commented on code/backend/test/test_attendees_success_cases.py:
```diff
@@ -0,0 +1,295 @@
+        "/attendees", params={"filter_expression": f"attendee_id:eq:{att_id}"}
+    )
+    assert after.status_code == 200
+    assert len(after.json()["items"]) == 0
```
nit: add new line at the end of the file

shreyarana7 replied:
uv run ruff format removes the new lines

shreyarana7 commented on code/backend/test/test_attendees_failure_cases.py:
```diff
@@ -0,0 +1,216 @@
+    r = await client.delete(f"/attendees/{uuid4()}")
+    assert r.status_code == 404
+    detail = r.json().get("detail", "")
+    assert detail and any(k in detail.lower() for k in ["no such", "not found", "does not"])
```
done
